packages/amplify-category-function/src/provider-utils/awscloudformation/utils/zipResource.ts

```diff
@@ -0,0 +1,31 @@
+import { ZipEntry } from 'amplify-function-plugin-interface';
+import * as fs from 'fs-extra';
+import archiver from 'archiver';
+
+export const zipPackage = (zipEntries: ZipEntry[], packageFileName: string): Promise<string> => {
+  if (zipEntries && zipEntries.length) {
+    const file = fs.createWriteStream(packageFileName);
+    const zip = archiver.create('zip', {});
+    return new Promise((resolve, reject) => {
+      file.on('close', () => {
+        resolve('Successfully zipped');
+      });
+      file.on('error', err => {
+        reject(new Error(`Failed to zip with error: [${err}]`));
+      });
+      zip.pipe(file);
```

Comment:
Is using `pipeline()` an option here? It is a bit better than `pipe()`. It also takes a callback, which we could use to `resolve()` or `reject()` a bit more cleanly (as the code is currently written, I *think* that both the `'close'` and `'error'` handler will execute if an error occurs).